### ISSUE TITLE
Default compression lz4 results initramfs too big

### ISSUE THREAD

**Issue by Xdavius** [Frogging-Family/linux-tkg#821]
Please,

Think about changing the default compression to zstd or xz to deal with 512mb archinstall boot partition 

Regards.

Davius

**Comment by Exotic0015** [comment]
I second this. My /boot partition cannot fit two linux-tkg kernels, so in order to update to a newer version I have to uninstall the one I have, install the stock kernel and while on the stock kernel install the new package

**Comment by Tk-Glitch** [comment]
On which OS @Exotic0015 ?
Because on Arch it takes pretty much the same space as a stock kernel, which is expected since we're using Arch's defconfig (zstd compressed image + initramfs). Modules aren't installed to the boot partition unless you manually add some to initramfs, which isn't any different than doing it with the stock kernel.

**Comment by Tk-Glitch** [comment]
Also as a side note, we're not using lz4 compression for anything but the Suse and generic kernel installation paths. For everything else it'll respect either the archlinux's defconfig (zstd everything) or your own config.

**Comment by Exotic0015** [comment]
Sorry, seems like that wasn't the issue. I'm on Fedora 39. The issue I'm getting is
```
grub2-mkrelpath: error: failed to get canonical path of `/boot/vmlinuz-6.5.5_tkg_bore_eevdf'.
dirname: missing operand
Try 'dirname --help' for more information.
```
Which at first I though was a matter of space, but in fact, it is not.

**Comment by AdelKS** [comment]
> Should I open a new issue for this?

Yeah sounds good, so at least the initial request can be found by others if someone else thinks the same.

**Comment by Tk-Glitch** [comment]
After deeper testing this isn't actually an issue. There is a different issue related to nvidia when adding their stuff to initramfs - which is what this issue was actually about - but we have nothing to do with it.

**Comment by ptr1337** [comment]
> After deeper testing this isn't actually an issue. There is a different issue related to nvidia when adding their stuff to initramfs - which is what this issue was actually about - but we have nothing to do with it.

That was a mkinitcpio issue using symlinks as normal path and putting them into the initramfs. Was fixed in mkinitcpio 37.2
